bdai_ros2_wrappers/test/test_integration.py

```diff
@@ -0,0 +1,162 @@
+        self._transform.transform.rotation.w = math.cos(math.radians(45) / 2)
+
+        self.callback()  # do not wait for first publish
+        self._timer = self.create_timer(1, self.callback)
+
+    def callback(self) -> None:
+        self._transform.header.stamp = self.get_clock().now().to_msg()
+        self._broadcaster.sendTransform(self._transform)
+
+
+class MinimalServer(FriendlyNode):
+    def __init__(self, node_name: str = "minimal_server", **kwargs: Any) -> None:
+        super().__init__(node_name, **kwargs)
+        self._service_server = self.create_service(AddTwoInts, "add_two_ints", self.service_callback)
+
+    def service_callback(self, request: AddTwoInts.Request, response: AddTwoInts.Response) -> AddTwoInts.Response:
+        response.sum = request.a + request.b
+        return response
+
+
+class MinimalActionServer(FriendlyNode):
+    def __init__(self, node_name: str = "minimal_action_server", **kwargs: Any) -> None:
+        super().__init__(node_name, **kwargs)
+        self._action_server = FriendlyActionServer(self, Fibonacci, "compute_fibonacci_sequence", self.execute_callback)
+
+    def execute_callback(self, goal_handle: ServerGoalHandle) -> Fibonacci.Result:
+        sequence = [0, 1]
+
+        for i in range(1, goal_handle.request.order):
+            sequence.append(sequence[i] + sequence[i - 1])
+            feedback = Fibonacci.Feedback()
+            feedback.partial_sequence = sequence
+            goal_handle.publish_feedback(feedback)
+        goal_handle.succeed()
+
+        result = Fibonacci.Result()
+        result.sequence = sequence
+        return result
+
+
+@pytest.fixture
+def ros_context() -> Generator[Context, None, None]:
+    context = Context()
+    rclpy.init(context=context)
+    try:
+        yield context
+    finally:
+        context.try_shutdown()
+
+
+@pytest.fixture
+def ros_executor(ros_context: Context) -> Generator[Executor, None, None]:
+    executor = AutoScalingMultiThreadedExecutor(context=ros_context)
+    try:
+        yield executor
+    finally:
+        executor.shutdown()
+
+
+@pytest.fixture
+def ros_node(ros_context: Context, ros_executor: Executor) -> Generator[Node, None, None]:
+    node = FriendlyNode("test_node", context=ros_context)
+    try:
+        yield node
+    finally:
+        node.destroy_node()
+
+
+@pytest.fixture(autouse=True)
+def ros_graph(ros_context: Context, ros_executor: Executor, ros_node: Node) -> Generator[None, None, None]:
+    ros_executor.add_node(MinimalTransformPublisher(context=ros_context))
+    ros_executor.add_node(MinimalServer(context=ros_context))
+    ros_executor.add_node(MinimalActionServer(context=ros_context))
+    ros_executor.add_node(ros_node)
+    try:
+        yield
+    finally:
+        for node in ros_executor.get_nodes():
+            ros_executor.remove_node(node)
+
+
+def test_blocking_sequence(ros_executor: Executor, ros_node: Node) -> None:
+    """
```

Comment:
FYI @jbarry-bdai @amessing-bdai is something like this that you had in mind when discussing a "simpler" programming model?

Reply:
Yes I think so.